src/combinatorics/quiver/symmetric.lean

```diff
@@ -0,0 +1,207 @@
+
+@[simp] lemma path.reverse_reverse [h : has_involutive_reverse V] {a b : V} (p : path a b) :
+  p.reverse.reverse = p :=
+begin
+  induction p,
+  { simp, },
+  { simp only [path.reverse, path.reverse_comp, path.reverse_to_path, reverse_reverse, p_ih],
+    refl, },
+end
+
+namespace symmetrify
+
+/-- The inclusion of a quiver in its symmetrification -/
+def of : prefunctor V (symmetrify V) :=
+{ obj := id,
+  map := λ X Y f, sum.inl f }
+
+/-- Given a quiver `V'` with reversible arrows, a prefunctor to `V'` can be lifted to one from
+    `symmetrify V` to `V'` -/
+def lift {V' : Type*} [quiver V'] [has_reverse V'] (φ : V ⥤q V') :
+  (symmetrify V) ⥤q V' :=
+{ obj := φ.obj,
+  map := λ X Y f, sum.rec (λ fwd, φ.map fwd) (λ bwd, reverse (φ.map bwd)) f }
+
+lemma lift_spec  (V' : Type*) [quiver V'] [has_reverse V'] (φ : V ⥤q V') :
+  of ⋙q (lift φ) = φ :=
+begin
+  fapply prefunctor.ext,
+  { rintro X, refl, },
+  { rintros X Y f, refl, },
+end
+
+lemma lift_reverse  (V' : Type*) [quiver V'] [h : has_involutive_reverse V']
+  (φ : V ⥤q V')
+  {X Y : symmetrify V} (f : X ⟶ Y) :
+  (lift φ).map (quiver.reverse f) = quiver.reverse ((lift φ).map f) :=
+begin
+  dsimp [lift], cases f,
+  { simp only, refl, },
+  { simp only, rw h.inv', refl, }
+end
+
+/-- `lift φ` is the only prefunctor extending `φ` and preserving reverses. -/
+lemma lift_unique (V' : Type*) [quiver V'] [has_reverse V']
+  (φ : V ⥤q V')
+  (Φ : (symmetrify V) ⥤q V')
+  (hΦ : of ⋙q Φ = φ) [hΦrev : Φ.map_reverse] :
+  Φ = lift φ :=
+begin
+  subst_vars,
+  fapply prefunctor.ext,
+  { rintro X, refl, },
+  { rintros X Y f,
+    cases f,
+    { refl, },
+    { dsimp [lift,of],
+      simp only [←prefunctor.map_reverse', symmetrify_reverse, sum.swap_inl], }, },
+end
+
+/-- A prefunctor canonically defines a prefunctor of the symmetrifications. -/
+@[simps] def _root_.prefunctor.symmetrify (φ : U ⥤q V) : (symmetrify U) ⥤q (symmetrify V) :=
+{ obj := φ.obj,
+  map := λ X Y, sum.map φ.map φ.map }
+
+instance _root_.prefunctor.symmetrify_map_reverse  (φ : U ⥤q V) :
+  prefunctor.map_reverse φ.symmetrify := ⟨λ u v e, by { cases e; refl }⟩
+
+end symmetrify
+
+namespace push
+
+variables {W} (σ : V → W)
+
+instance [has_reverse V] : has_reverse (push σ) :=
+{ reverse' := λ a b F, by { cases F, constructor, apply reverse, exact F_f, } }
+
+instance [h : quiver.has_involutive_reverse V] : quiver.has_involutive_reverse (push σ) :=
+{ reverse' := λ a b F, by { cases F, constructor, apply reverse, exact F_f, },
+  inv' :=  λ a b F, by { cases F, dsimp [reverse], congr, apply h.inv', } }
+
+@[simp] lemma of_reverse [h : has_involutive_reverse V]  (X Y : V) (f : X ⟶ Y):
+  (reverse $ ((of σ)).map f) = ((of σ)).map (reverse f) := rfl
+
+end push
+
+/--
+A quiver with involutive reverses is connected iff there exists a path between any pair of
+vertices.
+-/
+@[nolint unused_arguments]
+def is_connected (V) [quiver.{u+1} V] [has_involutive_reverse V] := ∀ (X Y : V), nonempty (path X Y)
```

Comment:
```suggestion
def is_connected (V) [quiver.{u+1} V] [has_reverse V] := ∀ (X Y : V), nonempty (path X Y)
```

Reply:
Actually, I'd say this definition doesn't require the `reverse` at all.

Reply:
I'd keep it as a token of the fact that the definition really takes its full sense when you can reverse arrows. Otherwise it still works but doesn't give you what you'd expect.

Reply:
In general there are two different notions: strong connectivity (where there should be a path from `X` to `Y`) and weak connectivity (where there should be a zigzag from `X` to `Y`). As soon as you have reversals, they are equivalent.

Also, to be consistent with the rest of mathlib, maybe this should be called `is_preconnected`? Connectivity should mean that also `nonempty V`

Reply:
Indeed, `is_preconnected` is better, although I'm tempted to simply drop this definition from the PR, since it's just sitting there with no actual use. 
Btw, @dwarn, I was thinking of maybe refactoring the `is_weakly_connected` code into `is_connected (symmetrify V)`. Would that make sense in your opinion?